lib/typing/mongo/bson.go

```diff
@@ -30,21 +31,36 @@ func JSONEToMap(val []byte) (map[string]any, error) {
 		return "null"
 	}))
 
-	var jsonMap map[string]any
 	var bsonDoc bson.D
 	err := bson.UnmarshalExtJSON(val, false, &bsonDoc)
 	if err != nil {
 		return nil, fmt.Errorf("failed to unmarshal ext json: %w", err)
 	}
 
-	bytes, err := bson.MarshalExtJSONWithRegistry(createCustomRegistry().Build(),
-		bsonDoc, false, true)
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewExtJSONValueWriter(buf, false, true)
```

Comment:
Could move the buffer, valuewriter, and encoder into `createCustomRegistry()` and rename it `createExtJSONEncoder()` or something. I also wonder if we'd get any performance gains by creating an encoder once and reusing it.

Reply:
I guess you'd have to pass the buffer into that method though.